addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/discovery/DSMRMeterDiscoveryService.java

```diff
@@ -78,27 +78,27 @@ public void telegramReceived(P1Telegram telegram) {
             logger.debug("Detect meters from #{} objects", telegram.getCosemObjects().size());
         }
         final Entry<Collection<DSMRMeterDescriptor>, Map<CosemObjectType, CosemObject>> detectedMeters = meterDetector
-                .detectMeters(telegram);
+            .detectMeters(telegram);
         verifyUnregisteredCosemObjects(telegram, detectedMeters.getValue());
         validateConfiguredMeters(dsmrBridgeHandler.getThing().getThings(),
-                detectedMeters.getKey().stream().map(md -> md.getMeterType()).collect(Collectors.toSet()));
+            detectedMeters.getKey().stream().map(md -> md.getMeterType()).collect(Collectors.toSet()));
         detectedMeters.getKey().forEach(m -> meterDiscovered(m, dsmrBridgeHandler.getThing().getUID()));
     }
 
     protected void verifyUnregisteredCosemObjects(P1Telegram telegram,
-            Map<CosemObjectType, CosemObject> undetectedCosemObjects) {
+        Map<CosemObjectType, CosemObject> undetectedCosemObjects) {
         if (!undetectedCosemObjects.isEmpty()) {
             if (undetectedCosemObjects.entrySet().stream()
-                    .anyMatch(e -> e.getKey() == CosemObjectType.METER_EQUIPMENT_IDENTIFIER
-                            && e.getValue().getCosemValues().entrySet().stream().anyMatch(
-                                    cv -> cv.getValue() instanceof StringType && cv.getValue().toString().isEmpty()))) {
+                .anyMatch(e -> e.getKey() == CosemObjectType.METER_EQUIPMENT_IDENTIFIER
+                    && e.getValue().getCosemValues().entrySet().stream()
+                        .anyMatch(cv -> cv.getValue() instanceof StringType && cv.getValue().toString().isEmpty()))) {
                 // Unregistered meter detected. log to the user.
                 reportUnregisteredMeters();
             } else {
                 reportUnrecognizedCosemObjects(undetectedCosemObjects);
-                logger.info(
-                        "There are some unrecognized values, which means some meters might not be detected. Not all values are recognized. Please report your raw data as example:",
-                        telegram.getRawTelegram());
+                logger.info("There are some unrecognized cosem values in the data received from the meter,"
+                    + " which means some meters might not be detected. Please report your raw data as reference: {}",
```

Comment:
Can we avoid using String concatenation?

Reply:
These are concatenations of static strings to improve readability. Otherwise it's a very long line. This is correct usage imo. The compiler will optimize the concatenation away so there is no technical reason to not do it like this.

Reply:
Thanks for refreshing my memory. I am reviewing too much PHP in my job.